--- Ice-3.4.2/py/modules/IcePy/Operation.cpp.str_fmt~	2012-02-17 03:52:30.788819095 +0100
+++ Ice-3.4.2/py/modules/IcePy/Operation.cpp	2012-02-17 03:53:17.754859121 +0100
@@ -1872,7 +1872,7 @@ IcePy::AsyncTypedInvocation::invoke(PyOb
         // IllegalArgumentException can propagate directly.
         // (Raised by checkAsyncTwowayOnly)
         //
-        PyErr_Format(PyExc_RuntimeError, STRCAST(ex.reason().c_str()));
+        PyErr_Format(PyExc_RuntimeError, "%s", STRCAST(ex.reason().c_str()));
         return 0;
     }
     catch(const Ice::Exception&)
@@ -1957,7 +1957,7 @@ IcePy::AsyncTypedInvocation::end(const I
     }
     catch(const IceUtil::IllegalArgumentException& ex)
     {
-        PyErr_Format(PyExc_RuntimeError, STRCAST(ex.reason().c_str()));
+        PyErr_Format(PyExc_RuntimeError, "%s", STRCAST(ex.reason().c_str()));
     }
     catch(const Ice::Exception& ex)
     {
@@ -2604,7 +2604,7 @@ IcePy::AsyncBlobjectInvocation::end(cons
     }
     catch(const IceUtil::IllegalArgumentException& ex)
     {
-        PyErr_Format(PyExc_RuntimeError, STRCAST(ex.reason().c_str()));
+        PyErr_Format(PyExc_RuntimeError, "%s", STRCAST(ex.reason().c_str()));
     }
     catch(const Ice::Exception& ex)
     {
--- Ice-3.4.2/rb/src/IceRuby/Operation.cpp.str_fmt~	2012-02-17 03:54:26.124917406 +0100
+++ Ice-3.4.2/rb/src/IceRuby/Operation.cpp	2012-02-17 03:54:44.893933408 +0100
@@ -240,7 +240,7 @@ IceRuby::OperationI::invoke(const Ice::O
 
     if(!_deprecateMessage.empty())
     {
-        rb_warning(_deprecateMessage.c_str());
+        rb_warning("%s", _deprecateMessage.c_str());
         _deprecateMessage.clear(); // Only show the warning once.
     }
 
